include/vcpkg/base/expected.h

```diff
@@ -105,191 +26,332 @@ namespace vcpkg
     template<class T>
     struct ExpectedHolder
```

Comment:
Since you're removing `ErrorHolder`, would it be "better" to change this to a traits struct as well? e.g.

```cpp
union {
  Error m_err;
  typename ExpectedHolder<T>::type m_t;
};
auto get() const noexcept { return ExpectedHolder<T>::get(m_t); }
```